### PR TITLE
Testing of RealmAdapter

### PR DESCRIPTION
**Pull request by benjaminprakash88**

### CODE REVIEW

**Comment by cmelchior** [comment]
Tests has been updated and should be ready for review.

@bmunkholm @emanuelez @kneth

File: realm/src/androidTest/java/io/realm/RealmAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    public void testUpdateRealmResultInAdapter() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        resultList.sort(FIELD_STRING);
+        RealmAdapter realmAdapter = new RealmAdapter(getContext(), resultList, false);
+        assertEquals(resultList.first().getColumnString(), realmAdapter.getRealmResults().first().getColumnString());
+        assertEquals(resultList.size(), realmAdapter.getRealmResults().size());
+
+        testRealm.beginTransaction();
+        AllTypes allTypes = testRealm.createObject(AllTypes.class);
+        allTypes.setColumnString("test data 47");
```
Comment (kneth) [inline realm/src/androidTest/java/io/realm/RealmAdapterTest.java]: IMHO, it is better to use `TEST_DATA_SIZE` than 47.

File: realm/src/androidTest/java/io/realm/RealmAdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    public void testClearFromAdapter() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        RealmAdapter realmAdapter = new RealmAdapter(getContext(), resultList, automaticUpdate);
+
+        testRealm.beginTransaction();
+        realmAdapter.getRealmResults().clear();
+        testRealm.commitTransaction();
+
+        assertEquals(0, realmAdapter.getCount());
```
Comment (kneth) [inline realm/src/androidTest/java/io/realm/RealmAdapterTest.java]: Should we test changes to `resultList` too?

Comment (cmelchior) [inline realm/src/androidTest/java/io/realm/RealmAdapterTest.java]: Yes, it has been added.

**Comment by emanuelez** [comment]
:+1:

**Pull request merged**